docs/tutorial/overview.md

```diff
@@ -28,33 +27,112 @@ component main() -> {
 }
 ```
 
-Once we parse our MrXL program into an [AST][astcode], we can decide on rules for generating code depending on which AST node we're working on. Depending on the AST node, we might need to add code to `cells`, `wires` or `control`.
+`cells` contains declarations for logical hardware units like adders, memories and registers. `wires` contains `group`s that connect together the units declared in `cell`s and form the structure of the hardware. `control` contains the logic specifying when the `group`s will perform their computation.
+
+First, we'll parse our program into an [AST][astcode]. This program:
+
+```
+input foo: int[4]
+output baz: int[4]
+baz := map 4 (a <- foo) { a + 5 }
+```
+
+has an AST that we could represent in Python like this:
+
+```
+Prog(
+  decls=[
+    Decl(input=True, name='foo', type=Type(base='Tree(int, [])', size=4)),
+    Decl(input=False, name='baz', type=Type(base='Tree(int, [])', size=4))
+  ],
+  stmts=[
+    Stmt(dest='baz',
+      op=Map(
+        par=4,
+        bind=[Bind(dest=['a'], src='foo')],
+        body=BinExpr(
+          op='add',
+          lhs=VarExpr(name='a'),
+          rhs=LitExpr(value=5)
+        )
+      )
+    )
+  ]
+)
+```
```

Comment:
I'm not sure if this is a useful thing to show at this point. Maybe you want to refer to some Python ASTs defined in the frontend. A cool feature of mdbook is [including files from source](https://rust-lang.github.io/mdBook/format/mdbook.html#including-files).